src/interface/report/Results/Timeline/Component.jsx

```diff
@@ -197,6 +244,54 @@ class Timeline extends PureComponent {
               '--cast-bars': castEvents.length,
             }}
           >
+            {ShowNPCSpellsToggle(
+              'Render NPC Spells on the timeline',
+              () => setRenderNPCSpells(!shouldRenderNPCSpells),
+              shouldRenderNPCSpells,
+            )}
+            {shouldRenderNPCSpells ? (
+              <>
+                {EnemySpellTypeToggle(
+                  'Instant Cast Abilities',
+                  () => this.handleToggle('instantCast'),
+                  instantCast,
+                )}
+                {EnemySpellTypeToggle(
+                  'channeled Abilities',
+                  () => this.handleToggle('channeledAbilities'),
+                  channeledAbilities,
+                )}
+                {EnemySpellTypeToggle(
+                  'Silenced/Interrupted Abilities',
+                  () => this.handleToggle('silencedInterrupted'),
+                  silencedInterrupted,
+                )}
+                {EnemySpellTypeToggle(
+                  'Boss Abilities',
+                  () => this.handleToggle('bossAbilities'),
+                  bossAbilities,
+                )}
```

Comment:
these should all be react components. to do this, you need to change the parameters to `EnemySpellTypeToggle` to be in an object:

```jsx
const EnemySpellTypeToggle = ({ label, toggleCallBack, checked }) => {
  // ...
};
```

and then replace these with:

```suggestion
<EnemySpellTypeToggle
  label={'Instant Cast Abilities'}
  toggleCallBack={() => this.handleToggle('instantCast')}
  checked={instantCast} />
<EnemySpellTypeToggle
  label={'Channeled Abilities'}
  toggleCallBack={() => this.handleToggle('channeledAbilities')}
  checked={channeledAbilities} />
<EnemySpellTypeToggle
  label={'Silenced/Interrupted Abilities'}
  toggleCallBack={() => this.handleToggle('silencedInterrupted')}
  checked={silencedInterrupted} />
<EnemySpellTypeToggle
  label={'Boss Abilities'}
  toggleCallBack={() => this.handleToggle('bossAbilities')}
  checked={bossAbilities} />
```

while this might seem like a style thing, there are subtle differences in how function call syntax vs component syntax is run and its better to use the more common style.